opensearch_py_ml/ml_commons/model_connector.py

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+from opensearchpy import OpenSearch
+
+from opensearch_py_ml.ml_commons.ml_common_utils import ML_BASE_URI
+
+
+class Connector:
+    def __init__(self, os_client: OpenSearch):
+        self.client = os_client
+
+    def create_standalone_connector(self, payload: dict):
+        if not isinstance(payload, dict):
+            raise ValueError("payload needs to be a dictionary")
+
+        return self.client.transport.perform_request(
+            method="POST", url=f"{ML_BASE_URI}/connectors/_create", body=payload
+        )
+
+    def list_connectors(self):
+        search_query = {"query": {"match_all": {}}}
+        return self.search_connectors(search_query)
+
+    def search_connectors(self, search_query: dict):
+        if not isinstance(search_query, dict):
+            raise ValueError("search_query needs to be a dictionary")
+
+        return self.client.transport.perform_request(
```

Comment:
Can we please check all the check warnings?

Reply:
Tests are acthally 100%. For somereason codecov is showing the stale results

Reply:
I'm sorry.  i probably got confused. What do you mean by a warning here? Did you meant the exception case?

Reply:
I meant whatever the codecode warning were showing.